docker-compose.yml

```
@@ -0,0 +1,13 @@
+version: '3.9'
+services:
+    web:
+        build: .
+        env_file:
+            - ./.env
+        ports:
+            - "5000:5000"
+        volumes:
+            - .:/todo_app
```

The WORKDIR specified in your Dockerfile is `/app`, not `/todo_app`. So this should be changed to `.:/app`

But also you could end up with errors sharing files/folders in the root of your project that you shouldn't, like `.venv`.

So I would change this to:
```suggestion
            - ./todo_app:/app/todo_app
```
Everything you want hotreloading for is in the todo_app folder.